Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -560,7 +560,7 @@ public void setLambda(double lambda) {
      * @return The aggregated and finalized result after sending a visitor through
      *         each tree in the forest.
      */
-    public <R, S> S traverseForest(double[] point, Function<ITree<?>, Visitor<R>> visitorFactory,
+    public <R, S> S traverseForest(double[] point, Function<ITree<?, ?>, Visitor<R>> visitorFactory,
```

Comment:
We could create a new `VisitorFactory<R>` interface to act as a typedef for this:
```
public interface VisitorFactor<R> extends Function<ITree<?, ?>, Visitor<R>> {}
```

Reply:
I like that.